Source/WebKit/UIProcess/ios/WKContentView.mm

```diff
@@ -232,12 +236,22 @@ - (void)_setupVisibilityPropagationViewForWebProcess
 {
     auto processIdentifier = _page->process().processIdentifier();
     auto contextID = _page->contextIDForVisibilityPropagationInWebProcess();
+#if HAVE(NON_HOSTING_VISIBILITY_PROPAGATION_VIEW)
+    if (!processIdentifier)
+#else
     if (!processIdentifier || !contextID)
+#endif
         return;
 
     ASSERT(!_visibilityPropagationViewForWebProcess);
     // Propagate the view's visibility state to the WebContent process so that it is marked as "Foreground Running" when necessary.
+#if HAVE(NON_HOSTING_VISIBILITY_PROPAGATION_VIEW)
+    auto environmentIdentifier = _page->process().environmentIdentifier();
+    auto viewClass = NSClassFromString(@"_UINonHostingVisibilityPropagationView");
```

Comment:
Do we need to do NSClassFromString if we're also doing a build-time version check? What's the case where we would build this code but not have the API at runtime?

Reply:
> Do we need to do NSClassFromString if we're also doing a build-time version check? What's the case where we would build this code but not have the API at runtime?

This was done to avoid compile errors in case one did not have the latest SDK, but is no longer needed. Fixed in latest patch.